### ISSUE TITLE
AnnotationValue.RenderingDispatcher.JAVA_14_CAPABLE_VM uses signed byte value for hex string

### ISSUE THREAD

**Issue by Marcono1234** [raphw/byte-buddy#1033]
`AnnotationValue.RenderingDispatcher.JAVA_14_CAPABLE_VM` uses the signed byte value for the hex string representation:
https://github.com/raphw/byte-buddy/blob/b175b6722f38fba61d20ba04429c262c1b73ea1f/byte-buddy-dep/src/main/java/net/bytebuddy/description/annotation/AnnotationValue.java#L195-L197

While the result is still correct, it is rather verbose:
```java
AnnotationValue.RenderingDispatcher.JAVA_14_CAPABLE_VM.toSourceString((byte) 0xFF);
// Result: (byte)0xffffffff
```

It would probably be better to use the unsigned byte value:
```java
return "(byte)0x" + Integer.toHexString(value & 0xFF);
// Result: (byte)0xff
```

**Comment by Marcono1234** [comment]
Which OpenJDK class is it whose behavior Byte Buddy tries to mimic?
Is it `sun.reflect.annotation.AnnotationInvocationHandler`? In that case it looks like it also uses two digits as well:
https://github.com/openjdk/jdk/blob/acf02ed5533e25ec572d18ff874a241860c66bed/src/java.base/share/classes/sun/reflect/annotation/AnnotationInvocationHandler.java#L302-L304

**Comment by raphw** [comment]
You are right, I fixed it on a branch where I currently work on annotations. Thanks for reporting.
